src/common/commonutils/UrlUtils.c

```diff
@@ -41,26 +71,23 @@ char* UrlEncode(char* target)
     return encodedTarget;
 }
 
-char* UrlDecode(char* target)
+char* UrlDecode(const char* target)
 {
-    int i = 0, j = 0;
+    size_t targetLength = 0, i = 0, j = 0;
     char buffer[3] = {0};
     unsigned int value = 0;
-    int targetLength = 0;
     char* decodedTarget = NULL;
 
-    if (NULL == target)
+    if ((NULL == target) || (0 >= (targetLength = strnlen(target, MAX_STRING_LENGTH))))
     {
         return NULL;
     }
 
-    targetLength = (int)strlen(target);
-    
     // The length of the decoded string is the same as the length of the encoded string or smaller
-    decodedTarget = (char*)malloc(targetLength + 1);
+    decodedTarget = (char*)malloc(targetLength + 3);

```

Comment:
why 2 more bytes?

Reply:
To allow room for movement towards the end of the buffer. See loop on line 92.